File: tests/testdata/codegen/aws-s3-logging-pp/nodejs/aws-s3-logging.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as aws from "@codeinfra/aws";

const logs = new aws.s3.Bucket("logs", {});
const bucket = new aws.s3.Bucket("bucket", {loggings: [{
    targetBucket: logs.bucket,
}]});
export const targetBucket = bucket.loggings.apply(loggings => loggings?.[0]?.targetBucket);

File: tests/testdata/codegen/aws-s3-logging-pp/python/aws-s3-logging.py
import codeinfra
import codeinfra_aws as aws

logs = aws.s3.Bucket("logs")
bucket = aws.s3.Bucket("bucket", loggings=[{
    "target_bucket": logs.bucket,
}])
codeinfra.export("targetBucket", bucket.loggings[0].target_bucket)
